codegenerator/cli/templates/static/codegen/src/EventFetching.res

```diff
@@ -88,7 +88,9 @@ let getTxFieldFromEthersLog = (log: Ethers.log, txField: string, ~logger): txFie
   }
 
 let transactionFieldsFromLog = (log, ~logger): Types.Transaction.t => {
-  Types.Transaction.fieldNames
+  //Note: if we implement all transaction fields, we will need all
+  //field names not just non optional ones
+  Types.Transaction.nonOptionalFieldNames
```

Comment:
Could you explain? I don't really understand the change. Is it related with rpc limitations?

Reply:
It's just that we don't fetch tx data from the RPC (yet). If we do it will be something similar to the lazy block loader for blocks. We currently just get the transactionIndex and transactionHash from the data that comes back in the getLogs request but we don't have anything more than that. This code works because txhash and txindex are non optional fields.

If you try and select any other fields while using an indexer with RPC it fails at validation during codegen.